supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesJobExecutionActor.scala

```diff
@@ -28,40 +33,59 @@ case class JesJobExecutionActor(override val jobDescriptor: BackendJobDescriptor
                                 serviceRegistryActor: ActorRef)
   extends BackendJobExecutionActor {
 
-  override def receive: Receive = LoggingReceive {
+  private def jesReceiveBehavior: Receive = LoggingReceive {
     case AbortJobCommand =>
       executor.foreach(_ ! AbortJobCommand)
     case abortResponse: AbortedResponse =>
       context.parent ! abortResponse
       context.stop(self)
-
-    // PBE TODO: use PartialFunction.orElse instead of this catch-all case, because Akka might use isDefinedAt over this partial function
-    case message => super.receive(message)
+    case KvPair(key, id @ Some(operationId)) if key.key == JesOperationIdKey =>
+      // Successful operation ID lookup during recover.
+      executor foreach { _ ! Recover(JesJobId(operationId))}
+    case KvKeyLookupFailed(_) =>
+      // Missed operation ID lookup during recover, fall back to execute.
+      executor foreach { _ ! Execute }
+    case KvFailure(_, e) =>
+      // Failed operation ID lookup during recover, crash and let the supervisor deal with it.
+      completionPromise.tryFailure(e)
+      throw new RuntimeException("Failure attempting to look up JES operation ID for key " + jobDescriptor.key, e)
   }
 
+  override def receive = jesReceiveBehavior orElse super.receive
+
   override val configurationDescriptor = jesConfiguration.configurationDescriptor
 
-  // PBE keep a reference to be able to hand to a successor executor if the failure is recoverable, or to complete as a
-  // failure if the failure is not recoverable.
   private lazy val completionPromise = Promise[BackendJobExecutionResponse]()
 
-  // PBE keep a reference for abort purposes.  Maybe we don't need this if we can look up actors by name.
   private var executor: Option[ActorRef] = None
 
-  // PBE there should be some consideration of supervision here.
-
-  override def recover: Future[BackendJobExecutionResponse] = ???
-
-  override def execute: Future[BackendJobExecutionResponse] = {
+  private def launchExecutor: Future[Unit] = Future {
     val executionProps = JesAsyncBackendJobExecutionActor.props(jobDescriptor,
       completionPromise,
       jesConfiguration,
       initializationData,
       serviceRegistryActor)
     val executorRef = context.actorOf(executionProps)
     executor = Option(executorRef)
-    executorRef ! Execute
-    completionPromise.future
+    ()
+  }
+
+  override def recover: Future[BackendJobExecutionResponse] = {
+    import JesJobExecutionActor._
+
+    for {
+      _ <- launchExecutor
+      _ = serviceRegistryActor ! KvGet(ScopedKey(jobDescriptor.descriptor.id, jobDescriptor.key, JesOperationIdKey))
+      c <- completionPromise.future
+    } yield c
```

Comment:
I would appreciate a tech talk (sort of like Khalid's database explanapaloosa) to explain what on earth is going on with these promises in this actor at some point.

E.g. why does a KvFailure above complete the promise with a failure but a KvSuccess does not, wouldn't the promise already have completed here? What even IS a promise? Much questions, many confusing.